napari/_vispy/canvas.py

```diff
@@ -1,63 +1,193 @@
 """VispyCanvas class.
 """
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
 from weakref import WeakSet
 
-from vispy.scene import SceneCanvas, Widget
+import numpy as np
+from vispy.scene import SceneCanvas as SceneCanvas_
+from vispy.scene import Widget
 
+from napari._vispy import VispyCamera
+from napari._vispy.utils.cursor import QtCursorVisual
 from napari._vispy.utils.gl import get_max_texture_sizes
+from napari._vispy.utils.visual import create_vispy_overlay
+from napari.components.overlays import CanvasOverlay, SceneOverlay
+from napari.utils._proxies import ReadOnlyWrapper
 from napari.utils.colormaps.standardize_color import transform_color
+from napari.utils.interactions import (
+    mouse_double_click_callbacks,
+    mouse_move_callbacks,
+    mouse_press_callbacks,
+    mouse_release_callbacks,
+    mouse_wheel_callbacks,
+)
+
+if TYPE_CHECKING:
+    from typing import Callable, List, Tuple, Union
+
+    import numpy.typing as npt
+    from qtpy.QtCore import Qt, pyqtBoundSignal
+    from qtpy.QtGui import QCursor, QImage
+    from vispy.app.canvas import DrawEvent, MouseEvent, ResizeEvent
+
+    from napari.components import ViewerModel
+    from napari.components.overlays import Overlay
+    from napari.utils.events.event import Event
+    from napari.utils.key_bindings import KeymapHandler
+
+
+class NapariSceneCanvas(SceneCanvas_):
```

Comment:
I'm not sure if this name came from another comment, but it's pretty odd to have `Napari` as the prefix for a class name in a napari package. I assume this is to distinguish it from the vispy scene canvas. How about importing that as `VispySceneCanvas` instead of `SceneCanvas_` and naming this `SceneCanvas`? Or just naming this `VispySceneCanvas` (since that's what it is other than the mouse event override) and not using an alias for `SceneCanvas`?

Reply:
It is harder to profile if you have multiple classes with the same name as you do not see if time is spent in napari class or base class.